### PR TITLE
0604_staging_PR

### LINKED ISSUE #305: Silver_JobStatus Cannot Merge field timeout_seconds -- casting string and long
Silver_JobStatus

May fail in some edge cases due to `timeout_seconds` originally being written as a string and later being written as a long. Delta will not implicitly cast from string to long.

Potential solution -- explicitly cast `timeout_seconds` to a long **but must validate downstream** that this doesn't cause schema issues for existing v6 schemas.

https://github.com/databrickslabs/overwatch/blob/ead79e15a5a6ffdd88d7ae7bbb05dd5775942fec/src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala#L1058

ERROR OUTPUT
```
FAILED --> 
ERROR:
Silver_JobsStatus FAILED -->

Failed to merge fields 'timeout_seconds' and 'timeout_seconds'. Failed to merge incompatible data types LongType and StringType
org.apache.spark.sql.AnalysisException: Failed to merge fields 'timeout_seconds' and 'timeout_seconds'. Failed to merge incompatible data types LongType and StringType
```

### LINKED ISSUE #310: Spark Job - several nulled fields -- version 0.6.0.3
Several fields like execution_id, notebook_id, job_id, etc are null in sparkJob table as of version 0.6.0.3

Cause: The schema cleanser was updated in 0603 to covert special characters to "" (originally) to "_" (0603). These map keys in the source had periods between words and thus the keys got converted to an _ instead of null string.

Solution: update the map keys to have the underscores.

### PR DESCRIPTION
**Pull request by GeekSheikh**
closes #305 
closes #309 
closes #310

### CODE REVIEW

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala [hunk 1]
```diff
@@ -483,4 +485,50 @@ object Helpers extends SparkSessionWrapper {
...
+    // verify database exists
+    assert(spark.catalog.databaseExists(etlDB), s"The database provided, $etlDB, does not exist.")
+    val dbMeta = spark.sessionState.catalog.getDatabaseMetadata(etlDB)
+    val dbProperties = dbMeta.properties
+
+    // verify database is owned and managed by Overwatch
+    assert(dbProperties.getOrElse("OVERWATCHDB", "FALSE") == "TRUE", s"The database provided, $etlDB, is not an Overwatch managed Database. Please provide an Overwatch managed database")
+    val workspaceID = if (dbutils.notebook.getContext.tags("orgId") == "0") {
+      dbutils.notebook.getContext.tags("browserHostName").split("\\.")(0)
+    } else dbutils.notebook.getContext.tags("orgId")
```
Comment (alexott) [inline src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala:506]: reuse function `getOrgId` defined earlier?

**Review by alexott — APPROVED** [review]
lgtm, just some comments

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala (hunk 1 above)
Comment (GeekSheikh) [reply src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala:506]: good one, thx, added it to next PR

File: src/main/scala/com/databricks/labs/overwatch/utils/SchemaScrubber.scala [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  /**
+   * Recursive function to drill into the schema. Currently only supports recursion through structs and array.
+   * TODO -- add support for recursion through Maps
+   * Issue_86
+   *
+   * @param dataType
+   * @return
+   */
+  private def sanitizeSchema(dataType: DataType, parentSanitizations: List[SanitizeRule] = List()): DataType = {
+    dataType match {
```
[the remark this answers is not in the document]
Comment (GeekSheikh) [reply src/main/scala/com/databricks/labs/overwatch/utils/SchemaScrubber.scala:107]: Not yet, but I can add that is on the TODO list as you can see above this function.

**Pull request merged**